### PR TITLE
feat(blacklist): populate blacklist while scanning for networks to avoid connect loop

### PR DESCRIPTION
**Pull request by comptonben**

### CODE REVIEW

File: uPy/main.py [hunk 1]
```diff
@@ -71,6 +71,11 @@ def sd_gdt_func(timer):
 wifi = "/sdcard/wifi.log"
 archive = "/sdcard/data.log"
 unsent = "/sdcard/buffer.log"
+blacklist = "/sdcard/blacklist.log"
+current_ap = "/sdcard/SSID.log"
+
+with open(blacklist, "a+"):
```
Comment (supersquirel500) [inline uPy/main.py]: why is this done this way over:  filepointer = open(blacklist, "a+")

File: uPy/main.py [hunk 2]
```diff
@@ -166,6 +177,11 @@ def sd_gdt_func(timer):
                 if posted:
                     remove(unsent)
 
+    with open(blacklist, 'r') as fp:
```
Comment (supersquirel500) [inline uPy/main.py]: why do we reopen the blacklist? Does this offer an advantage?

File: uPy/main.py (hunk 1 above)
Comment (DavidTou) [reply uPy/main.py]: Add comments Mr. Compton

Comment (DavidTou) [reply uPy/main.py]: Partial credit!!!

Comment (comptonben) [reply uPy/main.py]: add a comment explaining this

**Pull request merged**